### ISSUE TITLE
current status?

### ISSUE THREAD

**Issue by lclem** [tobiasBora/ogit_overleaf_v2_unofficial_git_bridge#1]
what is the status of this interesting project? is reverse sync "git --> overleaf/sharelatex" currently supported?

**Comment by tobiasBora** [comment]
Yes, both sync direction are supported, at least if the api didn't change meanwhile (I still want to add some commands to make it still easier to use). I also really need to write the documentation of the module.

Meanwhile, if you want to use the script, you can start by running the main python file with --help flag, it should give you the different commands. Just one thing: you should never go in the branch "overleaf". Modify master branch, and let the script synchronize the "overleaf" branch with both the online content and the master branch. In case of a conflict, you should fix it also in master branch.

Note however that this script is given "as it" and it could in theory remove all files on overleaf in case a strange bug appears (I had such error once due to binary upload, now the bug should be fixed). However, if such error happens, you should have zip backups that are dowloaded before any operation locally. But test the script on dummy project first !

Let me know if it works for you or if you have more questions.

Cheers,
tobiasBora

**Comment by tobiasBora** [comment]
Hum, I just wrote the README, but I tried the script on my phone and the last push command seems to fail. I'll try again at home and investigate.

**Comment by lclem** [comment]
Thanks for the clarifications, looking forward for updates :)

**Comment by tobiasBora** [comment]
So I confirm the project is not working from now (only for the direction local --> online)... I tried to investigate, and I still can't figure out why it does not manage to connect to the websocket... (or to be more precised the socket is closed with a message `7:::1+0`) I though it was because overleaf moved the server, and therefore I tried to use a session to make sure I have all the cookies linked with the google cloud stuff... But it still fails, while the headers are made super close to the one I have with Chromium, using something like:

```
websocat 'wss://fr.overleaf.com/socket.io/1/websocket/4XX_XXs0fRmlgnXLXXXX' -H 'Pragma: no-cache' -H 'Origin: https://fr.overleaf.com' -H 'Accept-Language: en-US,en;q=0.9' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.75 Safari/537.36' -H 'Upgrade: websocket' -H 'Cookie: gke-route=1568853647.714.43.XXXXXX;overleaf_session=s%somenumbers;' -H 'Connection: Upgrade'
```

If someone has some knowledge about `socket.io.js` I'd be curious to know what's could go wrong with that...

**Comment by lclem** [comment]
not sure whether this helps, but the [this](https://github.com/Jorl17/sharelatex-git-integration-unofficial) related project still works; it may be worth comparing them :)

**Comment by tobiasBora** [comment]
Thanks, but this project does not provide the link git —> overleaf, it is only overleaf —> git. And the only buggy part in my script is this direction, so this project can't help :-(

-- 
Tobias Bora
